src/fparser/two/Fortran2003.py

```diff
@@ -1501,161 +1506,294 @@ def match(string):
+                      or
+                      :py:class:`fparser.common.readfortran.FortranStringReader`
+        :return: code block containing instances of the classes that match
+                 the syntax of the type-bound procedure part of a derived type.
+        '''
         return BlockBase.match(Contains_Stmt,
                                [Binding_Private_Stmt, Proc_Binding_Stmt],
                                None, reader)
 
 
-class Binding_Private_Stmt(StmtBase, STRINGBase):  # R449
-    """
+class Binding_Private_Stmt(StmtBase, STRINGBase):
+    # pylint: disable=invalid-name
+    '''
+    :F03R:`449`::
+
+    Fortran 2003 rule R449
+    for binding private statement within the type-bound procedure
+    part of a derived type.
+
     <binding-private-stmt> = PRIVATE
-    """
+    '''
     subclass_names = []
 
+    @staticmethod
     def match(string):
+        '''
+        :param str string: Fortran code to check for a match
+        :return: keyword  "PRIVATE" or nothing if no match is found
+        :rtype: string
+        '''
         return StringBase.match('PRIVATE', string.upper())
-    match = staticmethod(match)
 
 
-class Proc_Binding_Stmt(Base):  # R450
-    """
+class Proc_Binding_Stmt(Base):
+    # pylint: disable=invalid-name
+    '''
+    :F03R:`450`::
+
+    Fortran 2003 rule R450
+    that specifies procedure binding for the type-bound procedures
+    within a derived type.
+
     <proc-binding-stmt> = <specific-binding>
                           | <generic-binding>
                           | <final-binding>
-    """
+    '''
     subclass_names = ['Specific_Binding', 'Generic_Binding', 'Final_Binding']
 
 
-class Specific_Binding(StmtBase):  # R451
-    """
+class Specific_Binding(StmtBase):
+    # pylint: disable=invalid-name
+    '''
+    :F03R:`451`::
+
+    Fortran 2003 rule R451
+    that specifies syntax of specific binding for a type-bound
+    procedure within a derived type.
+
     <specific-binding> = PROCEDURE [ ( <interface-name> ) ] [
         [ , <binding-attr-list> ] :: ] <binding-name> [ => <procedure-name> ]
-    """
+    '''
     subclass_names = []
     use_names = ['Interface_Name', 'Binding_Attr_List',
                  'Binding_Name', 'Procedure_Name']
 
     @staticmethod
     def match(string):
+        '''
+        :param str string: Fortran code to check for a match
+        :return: 5-tuple containing strings and instances of the classes
+                 describing a specific type-bound procedure (optional
+                 interface name, optional binding attribute list,
+                 optional double colon delimiter, mandatory binding
+                 name and optional procedure name)
+        :rtype: 5-tuple of objects (1 mandatory and 4 optional)
+        :raises NoMatchError: if the interface name is not preceded by '::'
+        '''
+        # Incorrect 'PROCEDURE' statement
         if string[:9].upper() != 'PROCEDURE':
```

Comment:
More checks need to be added here.

What if None is passed in?
What if the string is fewer than 9 characters long.
What happens if there is leading or trailing white space?